src/lib.rs

```diff
@@ -308,165 +173,288 @@ impl KafkaJsonToDelta {
         &mut self,
         cancellation_token: Option<&CancellationToken>,
     ) -> Result<(), KafkaJsonToDeltaError> {
-        let mut delta_table = deltalake::open_table(self.table_location.as_str()).await?;
-        let mut json_buffer: Vec<Value> = Vec::new();
-        let mut stream = self.consumer.stream();
-        let metadata = delta_table.get_metadata()?.clone();
-        let schema = metadata.schema.clone();
-        let mut delta_writer =
-            DeltaParquetWriter::for_table_path_with_schema(delta_table.table_path.clone(), &schema)
-                .await?;
+        let mut delta_writer = DeltaWriter::for_table_path(self.table_location.clone()).await?;
+        let mut value_buffers = ValueBuffers::new();
 
-        let arrow_schema_ref = Arc::new(<ArrowSchema as From<&Schema>>::from(
-            &metadata.clone().schema,
-        ));
-        let partition_cols = metadata.partition_columns;
+        let transformer = Transformer::from_transforms(&self.transforms)?;
+        let wal = write_ahead_log::new_write_ahead_log(self.app_id.to_string()).await?;
 
-        let mut latency_timer = Instant::now();
+        info!("Initializing last Delta transaction version before starting run loop.");
 
-        let expressions = compile_transforms(&self.transforms)?;
+        let last_txn_version = delta_writer
+            .last_transaction_version(self.app_id.to_string().as_str())
```

Comment:
If I'm reading this right, isn't `self.app_id` already `String`? Is the `to_string()` being a lazy `.clone()` here or is it just unnecessary?

Reply:
Nice catch. Unnecessary. Removed.